src/components/primitive/box/box.component.tsx

```diff
@@ -146,6 +149,19 @@ export const Box: React.FunctionComponent<BoxComponentProps> = (
     return <></>;
   }
 
+  if (loading) {
+    return (
+      <Base style={style} {...rest}>
```

Comment:
This line looks the same as below. Why have you used a separate return statement rather than using conditional rendering in the existing return statement?

Reply:
Unless this approach it's not working